### PR TITLE
✅ Improve amp-date-picker coverage with state machine and dates-list

### PR DESCRIPTION
**Pull request by cvializ**
Improve coverage on amp-date-picker.

I found and fixed a contract bug in dates-list 😅

### CODE REVIEW

**Comment by cvializ** [comment]
consider this in draft for the moment

**Comment by cvializ** [comment]
Ready for review!

File: extensions/amp-date-picker/0.1/test/test-amp-date-picker-state.js [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+describes.realWin(
+  'amp-date-picker',
+  {
+    amp: {
+      runtimeOn: false,
+      extensions: ['amp-date-picker'],
+    },
+  },
+  env => {
+    // const moment = requireExternal('moment');
```
Comment (sparhami) [inline extensions/amp-date-picker/0.1/test/test-amp-date-picker-state.js]: still needed?

File: extensions/amp-date-picker/0.1/test/test-amp-date-picker-state.js [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+    let clock;
+    let document;
+
+    const DEFAULT_ATTRS = {
+      layout: 'fixed-height',
+      height: '360',
+    };
+
+    function createDatePicker(opt_attrs, opt_parent = document.body) {
```
Comment (sparhami) [inline extensions/amp-date-picker/0.1/test/test-amp-date-picker-state.js]: seems like you should need a jsdoc comment for this function?

File: extensions/amp-date-picker/0.1/test/test-amp-date-picker-state.js [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+    });
+
+    describe('static states', () => {
+      it('should start in the static state', () => {
+        const {picker, layoutCallback} = createDatePicker({
+          'layout': 'fixed-height',
+          'height': '360',
+        });
+
+        return layoutCallback().then(() => {
```
Comment (sparhami) [inline extensions/amp-date-picker/0.1/test/test-amp-date-picker-state.js]: async/await?

**Pull request merged**